File: src/domain/models/UserPreferences.ts
import { WeightUnit } from './Models';

/**
 * User preferences data
 */
export interface UserPreferences {
  weightUnit?: WeightUnit;
  autoplayEnabled?: boolean;
  stopAtTop?: boolean; // false = stop at bottom (extended), true = stop at top (contracted)
}

/**
 * Create default user preferences
 */
export const createDefaultUserPreferences = (): UserPreferences => ({
  weightUnit: WeightUnit.KG,
  autoplayEnabled: true,
  stopAtTop: false,
});

File: deprecated/app/src/main/java/com/example/vitruvianredux/domain/model/UserPreferences.kt
package com.example.vitruvianredux.domain.model

/**
 * User preferences data class
 */
data class UserPreferences(
    val weightUnit: WeightUnit = WeightUnit.KG,
    val autoplayEnabled: Boolean = true,
    val stopAtTop: Boolean = false  // false = stop at bottom (extended), true = stop at top (contracted)
)
